PublicMapping/districtbuilder
# Add button to competitiveness page for uploading a plan to PlanScore

## Overview

- Add button to competitiveness page for uploading a plan to PlanScore
- Add new endpoint to upload to PlanScore API from backend
- Open PlanScore in new window when API response received

### Checklist


### Demo

![image](https://user-images.githubusercontent.com/66973361/125352792-b2f06880-e32f-11eb-97ff-45c8e386ea46.png)
![image](https://user-images.githubusercontent.com/66973361/125352862-cd2a4680-e32f-11eb-97bb-c2bcd6004dd7.png)
![image](https://user-images.githubusercontent.com/66973361/125353469-9a348280-e330-11eb-9ac2-54535ccce3d7.png)


### Notes

- The "Open in new tab" suggestion feels a bit clunky to me. By default, Chrome seems to block the popup for me on localhost unless I explicitly allow it. Personally, I might prefer a link to be displayed in the sidebar or in a new modal allowing the user to open the PlanScore result in a new tab at their leisure.

- it seems like bad etiquette for us to allow users to ping PlanScore's API multiple times for a district plan that has not changed. One potential idea I had to rectify this would be to store the `plan_score_url` received from the API and the `plan_score_last_updated_dt` of when the PlanScore was last computed, and to disable the 'Send to PlanScore' button unless the Project's `updatedDt` is more recent than the last time PlanScore was checked. 

- It might also be nice to add some feedback that the PlanScore API is processing the request - we could use the existing spinner from the project, but I'm curious to see if @jfrankl has any other ideas

## Testing Instructions

- `./scripts/update`
- Open a finished project and navigate to the Competitiveness metric page
- Click the `Send to PlanScore API` button and wait for a new window to appear. Note that you may need to enable popups on localhost for the popup to appear
- Expect: A new window indicating successful upload to PlanScore appears
- Wait a minute and refresh the PlanScore window - expect the PlanScore report for the district to be computed

Closes #838
